application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java

```diff
@@ -487,6 +495,110 @@ public JdaTester() {
         return ErrorResponseException.create(reason, new Response(null, -1, "", -1, Set.of()));
     }
 
+    /**
+     * Creates a MessageReceivedEvent
+     * <p>
+     * The method creates a MessageReceivedEvent from the input message and mock the message author
+     * from the input user.
+     *
+     * @param message the message to create an event for
+     * @param user the author from the message
+     * @return the created MessageReceivedEvent
+     */
+    public MessageReceivedEvent createMessageReceivedEvent(@NotNull Message message,
+            @NotNull User user) {
+        doReturn(user).when(message).getAuthor();
+        return new MessageReceivedEvent(jda, MESSAGE_RESPONSE_NUMBER, message);
+    }
+
+    /**
+     * Creates a message spy
+     *
+     * @return the created spy
+     */
+    public @NotNull Message createMessageSpy() {
+        return spy(Message.class);
+    }
+
+    /**
+     * Creates an attachment
+     *
+     * @param id id of the attachment
+     * @param url url of the attachment
+     * @param proxyUrl proxyUrl of the attachment
+     * @param fileName fileName of the attachment
+     * @param contentType contentType of the attachment
+     * @param description description of the attachment
+     * @param size size of the attachment
+     * @param height height of the attachment
+     * @param width width of the attachment
+     * @param ephemeral is the attachment ephemeral
+     * @return the created attachment
+     */
+    public Message.Attachment createAttachment(long id, String url, String proxyUrl,
+            String fileName, String contentType, String description, int size, int height,
+            int width, boolean ephemeral) {
+        return new Message.Attachment(id, url, proxyUrl, fileName, contentType, description, size,
+                height, width, ephemeral, jda);
+    }
+
+    /**
+     * Mocks the attachments from the given message
+     *
+     * @param message the message with the mocked attachments
+     * @param attachments the attachments which the mock should return
+     */
+    public void mockMessageAttachments(Message message, List<Message.Attachment> attachments) {
+        when(message.getAttachments()).thenReturn(attachments);
+    }
```

Comment:
I find this method super confusing. I would move all of this to the utility that u want to provide for message-creation. So upwards to the `MessageBuilder` thingy.

Reply:
MessageBuilder doesnt work with new MessageReceivedEvent thats why i created a spy, but still i think it would be better if you could look into the JdaTester part, i try to close the other open conversations.